File: analysis.py
def calculate_salary(hours, rate):
    base = hours * rate
    tax = base * 0.1
    return base - tax

def process_data(data):
    cleaned = [d.upper() for d in data if d]
    return sorted(cleaned)

def generate_report(name, score):
    if score >= 85:
        grade = "Distinction"
    else:
        grade = "Pass"
    return {"name": name, "grade": grade}

def display_summary(data):
    print("=== REPORT ===")
    for idx, item in enumerate(data, start=1):
        print(f"{idx}. {item}")

File: lib/analysis.js
// Converted utilities from analysis.py to JavaScript for use in Next.js
export function calculateSalary(hours, rate) {
  const total = hours * rate;
  const bonus = 500;
  return total + bonus;
}

export function processData(data) {
  const cleaned = data.map((d) => String(d).trim().toLowerCase());
  const unique = Array.from(new Set(cleaned));
  return unique;
}

export function generateReport(name, score) {
  const grade = score > 90 ? 'A' : 'B';
  return `Student: ${name}, Grade: ${grade}`;
}

export function displaySummary(data) {
  console.log('Summary Report:');
  data.forEach((item) => console.log(`- ${item}`));
}
